pkg/mimirpb/timeseries.go

```diff
@@ -590,3 +590,40 @@ func copyToYoloString(buf []byte, src string) (string, []byte) {
 	copy(buf, *((*[]byte)(unsafe.Pointer(&src))))
 	return yoloString(buf), buf[len(buf):]
 }
+
+// ForIndexes builds a new WriteRequest from the given WriteRequest, containing only the timeseries and metadata for the given indexes.
+// It assumes the indexes before the initialMetadataIndex are timeseries, and the rest are metadata.
+func (p *WriteRequest) ForIndexes(indexes []int, initialMetadataIndex int) *WriteRequest {
+	var timeseriesCount, metadataCount int
+	for _, i := range indexes {
+		if i >= initialMetadataIndex {
+			metadataCount++
+		} else {
+			timeseriesCount++
+		}
+	}
```

Comment:
can't we do this with binary search?

Reply:
We could, but it's not necessarily faster. We'd need a benchmark, and I need to stop somewhere :D